lib/expression/eval.ex

```diff
@@ -177,7 +177,7 @@ defmodule Expression.Eval do
   def op(:==, a, b) when is_struct(a, DateTime) and is_struct(b, DateTime),
     do: DateTime.compare(a, b) == :eq
 
-  def op(:=, a, b) when is_struct(a, Date) and is_struct(b, Date),
+  def op(:=, a, b) when is_struct(a, DateTime) and is_struct(b, DateTime),
```

Comment:
I think this was a typo, as this function clause was appearing twice.
I think this one was supposed to be for `DateTime`, not for `Date`.

Reply:
@smn could you confirm?

Reply:
yeah, most probably